receiver/vcenterreceiver/internal/mockserver/client_mock.go

```diff
@@ -0,0 +1,255 @@
+
+	case content == "group-h5" && contentType == "Folder":
+		if propSetArray {
+			arr := specSet["propSet"].([]interface{})
+			for _, i := range arr {
+				m, ok := i.(map[string]interface{})
+				require.True(t, ok)
+				if m["type"] == "ClusterComputeResource" {
+					return loadResponse("host-cluster.xml")
+				}
+			}
+		}
+		return loadResponse("host-parent.xml")
+
+	case content == "datastore-1003" && contentType == "Datastore":
+		if objectSetArray {
+			return loadResponse("datastore-list.xml")
+		}
+		return loadResponse("datastore-summary.xml")
+
+	case contentType == "HostSystem":
+		if ps, ok := propSet["pathSet"].([]interface{}); ok {
+			for _, v := range ps {
+				if v == "summary.hardware" {
+					return loadResponse("host-properties.xml")
+				}
+			}
+		} else {
+			ps, ok := propSet["pathSet"].(string)
+			require.True(t, ok)
+			if ps == "name" {
+				return loadResponse("host-names.xml")
+			}
+
+		}
+
+	case content == "group-v4" && contentType == "Folder":
+		if propSetArray {
+			return loadResponse("vm-group.xml")
+		}
+		if propSet == nil {
+			return loadResponse("vm-folder.xml")
+		}
+		return loadResponse("vm-folder-parent.xml")
+
+	case content == "vm-1040" && contentType == "VirtualMachine":
+		if propSet["pathSet"] == "summary.runtime.host" {
+			return loadResponse("vm-host.xml")
+		}
+		return loadResponse("vm-properties.xml")
+
+	case (content == "group-v1034" || content == "group-v1001") && contentType == "Folder":
+		return loadResponse("vm-empty-folder.xml")
+
+	case contentType == "ResourcePool":
+		if ps, ok := propSet["pathSet"].([]interface{}); ok {
+			for _, prop := range ps {
+				if prop == "summary" {
+					return loadResponse("resource-pool-summary.xml")
+				}
+			}
+		}
+
+		if ss, ok := objectSet["selectSet"].(map[string]interface{}); ok && ss["path"] == "resourcePool" {
+			return loadResponse("resource-pool-group.xml")
+		}
+
+	case objectSetArray:
+		objectArray := specSet["objectSet"].([]interface{})
+		for _, i := range objectArray {
+			m, ok := i.(map[string]interface{})
+			require.True(t, ok)
+			mObj := m["obj"].(map[string](interface{}))
+			typeString := mObj["-type"]
+			if typeString == "HostSystem" {
+				return loadResponse("host-names.xml")
+			}
+		}
+	}
+
+	return []byte{}, errNotFound
+}
+
+func routePerformanceQuery(t *testing.T, body map[string]interface{}) ([]byte, error) {
+	queryPerf := body["QueryPerf"].(map[string]interface{})
+	require.NotNil(t, queryPerf)
+	querySpec := queryPerf["querySpec"].(map[string]interface{})
+	entity := querySpec["entity"].(map[string]interface{})
+	switch entity["-type"] {
+	case "HostSystem":
+		return loadResponse("host-performance-counters.xml")
+	case "VirtualMachine":
+		return loadResponse("vm-performance-counters.xml")
+	}
+	return []byte{}, errNotFound
+}
+
+func loadResponse(filename string) ([]byte, error) {
+	return ioutil.ReadFile(filepath.Join("internal", "mockserver", "responses", filename))
```

Comment:
Nice job capturing all this.

Is it possible to document how these responses were captured, in case they need to be updated at some point/